CMakeLists.txt

```diff
@@ -52,15 +54,24 @@ else()
     set(CMAKE_INTERPROCEDURAL_OPTIMIZATION FALSE)
 endif()
 
-if (WIN32 OR EXTERNAL_CHIPDB)
+check_cxx_compiler_hash_embed(HAS_HASH_EMBED CXX_FLAGS_HASH_EMBED)
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${CXX_FLAGS_HASH_EMBED}")
+if (EXTERNAL_CHIPDB)
     set(BBASM_MODE "binary")
+elseif (HAS_HASH_EMBED)
+    set(BBASM_MODE "embed")
+elseif (WIN32)
+    set(BBASM_MODE "resource")
 else()
     set(BBASM_MODE "string")
 endif()
 
 find_package(Threads)
 if (Threads_FOUND)
-    find_package(TBB QUIET)
+    if (NOT STATIC_BUILD)
+        # TBB is not provided as a statically linked library.
+        find_package(TBB QUIET)
```

Comment:
as the PR is touching this anyway, might as well remove the optional TBB dependency, it was only used by the now-removed FPGA interface support

Reply:
done